.github/workflows/non-release.yaml

```diff
@@ -0,0 +1,26 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
```

Comment:
We missed this. The default branch name for this repo is `main`, not `master`. 🙈 
I raised https://github.com/gardener/gardener-extension-registry-cache/pull/443.